_posts/2019-02-04-lecture-06.md

```diff
@@ -0,0 +1,503 @@
+ 
+- Posterior *expert responsibilities* assigning points to experts are 
+
+$$p(z^k=1 \mid x,y,\theta) = \frac{p(z^k=1 \mid x) p_k(y \mid x, \theta_k, \sigma_k^2)}{\sum_j p(z^j=1 \mid x) p_j(y \mid x, \theta_j, \sigma_j^2)}$$
+
+- Using the expected complete log likelihood as loss
+
+$$
+\begin{aligned}
+\langle \ell_c(\theta; x,y,z) \rangle &= \sum_n \langle \log p(z_n \mid x_n, \chi) \rangle_{p(z \mid x, y)} + \sum_n \langle \log p(y_n \mid x_n, z_n, \theta, \sigma) \rangle_{p(z \mid x,y)} \\
+&= \sum_n \sum_k \langle z_n^k \rangle \log \big( \operatorname{softmax}(\chi_k^Tx_n)\big) - \frac{1}{2} \sum_n \sum_k \langle z_n^k \rangle \Big( \frac{y_n-\theta_k^Tx_n}{\sigma_k^2} + \log \sigma_k^2 + C \Big)
+\end{aligned}
+$$
+
+The resulting EM algorithm:
+
+- E-step calculates the responsibilities $$\tau_n^{k(t)}=p(z_n^k=1 \mid x_n,y_n, \theta) = \frac{p(z_n^k=1 \mid x_n)p_k(y_n \mid x_n, \theta_k, \sigma_k^2)}{\sum_j p(z_n^j=1 \mid x_n)p_j(y_n \mid x_n, \theta_j, \sigma_j^2)}$$
+- M-step calculates linear regressions for each expert, each data point weighted by the expert's responsibility **(see homework)** 
+
+#### EM Variants
++ **Partially Hidden Variables**: We can use EM when there are missing (hidden) variables in some cases and not in others. In the E-Step we can estimate the hidden variables for only the incomplete cases. The M-step then optimizes the log-likelihood on the complete data plus the expected likelihood of the incomplete data found using the E-step.
++ **Sparse EM**: In this variant, you do not exactly re-compute the posterior probability on each data point under all models, because it is practically zero. Instead, you keep an "active list" which you update every once in a while.
++ **Generalized (Incomplete) EM**: It might be hard to find the ML parameters in the M-step, even given the completed data. We can still make progress by doing an M-step that improves the likelihood a bit (e.g. gradient step). Recall the IRLS step in the mixture of experts model.
+
+## Summary
+
+Learning the parameters of **partially observed graphical models**, or latent variable models, is considerably more difficult than for fully observed GMs. This is because the unobserved or latent variables tie the observed variables together via marginalization, and we can no longer decompose the maximum likelihood function into independent conditional probabilities.
+
+The **EM Algorithm** provides a way of maximizing the likelihood function for latent variable models. We can find the MLE parameters when the original problem can be broken into 2 steps: (1) Estimating or "filling in" the unobserved latent variables using the best guess (posterior) provided from our observed data and current parameters, and (2) updating the parameters based on this guess:
+
+**E-Step**
+
+$$
+q^{(t+1)} = \text{arg max}_q F(q, \theta^t)
+$$
+
+**M-Step**
+
+$$
+\theta^{(t+1)} = \text{arg max} _{ \theta} F(q^{(t+1)}, \theta ^t)
+$$
+
++ **EM Pros**:
```

Comment:
these are important learning points, can you explain them in more detail? thanks!